embulk-core/src/main/java/org/embulk/exec/BulkLoader.java

```diff
@@ -125,12 +142,25 @@ public void setFilterTaskSources(List<TaskSource> filterTaskSources)
             this.filterTaskSources = filterTaskSources;
         }
 
+        public void setReporterTaskSources(Map<Reporter.Channel, TaskSource> reporterTaskSources)
+        {
+            this.reporterTaskSources = reporterTaskSources;
+        }
+
         public ProcessTask buildProcessTask()
         {
             return new ProcessTask(
-                    plugins.getInputPluginType(), plugins.getOutputPluginType(), plugins.getFilterPluginTypes(),
-                    inputTaskSource, outputTaskSource, filterTaskSources,
-                    schemas, executorSchema, Exec.newTaskSource());
+                    plugins.getInputPluginType(),
+                    plugins.getOutputPluginType(),
+                    plugins.getFilterPluginTypes(),
+                    reporterPluginTypes,
```

Comment:
Any reasons to have this as a field, not like other plugin's `plugins.getInputPluginType()`? If it has special reasons, we may want to leave comments.